mobilecoind/src/bin/main.rs

```diff
@@ -27,8 +27,14 @@ fn main() {
     let _sentry_guard = mc_common::sentry::init();
     let (logger, _global_logger_guard) = create_app_logger(o!());
 
-    let _tracer = mc_util_telemetry::setup_default_tracer(env!("CARGO_PKG_NAME"))
-        .expect("Failed setting telemetry tracer");
+    // Telemetry is disabled if MC_TELEMETRY is set to "0"
+    let telemetry_disabled = std::env::var("MC_TELEMETRY")
+        .map(|val| val == "0")
+        .unwrap_or(false);
```

Comment:
maybe this test should actually go in `mc_util_telemetry::setup_default_tracer` so that all the binaries get it

Reply:
I think that is probably better. I think it can be changed to return a `Result<Option<sdk::trace::Tracer>, Error>` to handle cases where telemetry is disabled, or perhaps we can provide our own implementation of `Tracer` that is no-op when disabled.

Reply:
My feeling is that if we're going to plumb this into every app, we should add a config object to `mc_util_telemetry` that every app includes (with `#[clap(flatten)]`), and make `setup_default_tracer` take that object instead.

Reply:
I am not sure what that gives us, since `setup_default_tracer` is already called in every app that wants to support telemetry. Changing `setup_default_tracer` would mean it gets automatically resolved everywhere (since the return value is only used to keep the tracer in scope, changing to `Option<>` would not mandate changing anything in the apps themselves).

Reply:
It keeps our run-time configuration consistently accessible either via command-line or environment variables.

Reply:
So then you also want to change all the `mc-common` logging stuff this way as well I presume?

Reply:
I believe this is already implicitly tested since there is code that does tracing that gets exercised in unit tests, and they do not setup a tracer.